crates/ruff/src/rules/pyupgrade/rules/f_strings.rs

```diff
@@ -247,17 +247,33 @@ pub(crate) fn f_strings(checker: &mut Checker, summary: &FormatSummary, expr: &E
 
     // Currently, the only issue we know of is in LibCST:
     // https://github.com/Instagram/LibCST/issues/846
-    let Some(contents) = try_convert_to_f_string(checker, expr) else {
+    let Some(mut contents) = try_convert_to_f_string(checker, expr) else {
         return;
     };
 
     // Avoid refactors that increase the resulting string length.
-    let existing = checker.locator.slice(expr);
-    if contents.len() > existing.len() {
-        return;
+    let expr_range = Range::from(expr);
+    if expr_range.location.column() == 0 {
+        let existing = checker.locator.slice(expr_range);
+        if contents.len() > existing.len() {
+            return;
+        }
+    } else {
+        let existing = checker.locator.slice(Range::new(
+            expr_range.location.with_col_offset(-1),
+            expr_range.end_location,
+        ));
+        if contents.len() > existing.len() - 1 {
+            return;
+        }
+        // Add a space if there is none between a keyword and the string.
+        // `return`, `yield`, `assert`, …
```

Comment:
There might be more cases to consider here -- I haven't really thought it through rigorously... But for example, `rf"{expr}"` is valid.

Reply:
But I don't think we can _just_ check `r`, because some keywords end in `r`.

Reply:
`rf"{expr}"` is inside the range of the expression, it the character before the expression that is looked at, not `r` from the string.
Also it is not possible to have `r` before the string in this case as it would be a syntax error `returnr"foo"`.

Reply:
There is also `and`, `or`, `not`, ...
That is why I used `is_ascii_alphabetic`.

Reply:
Ahh right right.

And this is valid, right?

```py
return\
f"foo"
```

Reply:
```python
def f():
    return\
"{}".format(1)
```
becomes
```python
def f():
    return\
f"{1}"
```